src/Raven.Client/Documents/Session/IDocumentQueryBase.cs

```diff
@@ -440,6 +440,36 @@ public interface IFilterDocumentQueryBase<T, TSelf> where TSelf : IDocumentQuery
         /// <inheritdoc cref="MoreLikeThisBase"/>
         /// <param name="moreLikeThis">Specified MoreLikeThisQuery.</param>
         TSelf MoreLikeThis(MoreLikeThisBase moreLikeThis);
+        
+        /// <summary>
+        /// Performs vector search on text data, by queried text.
+        /// </summary>
+        /// <param name="textFieldFactory">Factory creating textual vector field for indexing purposes.</param>
+        /// <param name="textValueFactory">Factory preparing queried data to be used in vector search.</param>
+        /// <param name="minimumSimilarity">Minimum similarity between queried text and text stored in a document to be matched by the query.</param>
+        /// <param name="numberOfCandidates">Number of candidate nodes for HNSW algorithm. The bigger the value, the more accurate search will be performed for the cost of more computation.</param>
+        /// <param name="isExact">Defines whether vector search will be performed in approximate or exact manner.</param>
+        TSelf VectorSearch(Func<IVectorFieldFactory<T>, IVectorEmbeddingTextField> textFieldFactory, Action<IVectorEmbeddingTextFieldValueFactory> textValueFactory, float minimumSimilarity = Constants.VectorSearch.DefaultMinimumSimilarity, int numberOfCandidates = Constants.VectorSearch.DefaultNumberOfCandidatesForQuerying, bool isExact = Constants.VectorSearch.DefaultIsExact);
+
+        /// <summary>
+        /// Performs vector search on embedding data, by queried embedding.
+        /// </summary>
+        /// <param name="embeddingFieldFactory">Factory creating embedding vector field for indexing purposes.</param>
+        /// <param name="embeddingValueFactory">Factory preparing queried data to be used in vector search.</param>
+        /// <param name="minimumSimilarity">Minimum similarity between queried embedding and embedding stored in a document to be matched by the query.</param>
+        /// <param name="numberOfCandidates">Number of candidate nodes for HNSW algorithm. The bigger the value, the more accurate search will be performed for the cost of more computation.</param>
+        /// <param name="isExact">Defines whether vector search will be performed in approximate or exact manner.</param>
+        TSelf VectorSearch(Func<IVectorFieldFactory<T>, IVectorEmbeddingField> embeddingFieldFactory, Action<IVectorEmbeddingFieldValueFactory> embeddingValueFactory, float minimumSimilarity = Constants.VectorSearch.DefaultMinimumSimilarity, int numberOfCandidates = Constants.VectorSearch.DefaultNumberOfCandidatesForQuerying, bool isExact = Constants.VectorSearch.DefaultIsExact);
+        
+        /// <summary>
+        /// Performs vector search on existing vector field.
```

Comment:
@Lwiel 
maybe we should add some info about index, because it may be misleading with document's field.